app/models/piece.rb

```diff
@@ -1,3 +1,81 @@
 class Piece < ActiveRecord::Base
   belongs_to :game
+
+  def check_square(x, y)
+    return true if @game.pieces.where(row_coordinate: x, column_coordinate: y).any?
+  end
+
+  def check_horizontal(s_x, _d_x, s_y, d_y)
+    strt = [s_y, d_y].min + 1
+    fin = [s_y, d_y].max - 1
+
+    while strt <= fin
+      return true if check_square(s_x, strt)
+      strt += 1
+    end
+  end
+
+  def check_vertical(s_x, d_x, s_y, _d_y)
```

Comment:
For these methods, since you're inside a `Piece` object. You don't need to pass around the current row/column. They are available in `row_coordinate` and `column_coordinate`